Sources/TuistSupport/Utils/FileHandler.swift

```diff
@@ -224,6 +225,43 @@ public class FileHandler: FileHandling {
         try path.throwingGlob(glob)
     }
 
+    public func throwingGlobExcludingOpaqueDirectories(_ path: AbsolutePath, glob: String) throws -> [AbsolutePath] {
+        try throwingGlob(path, glob: glob)
+            .filter { !Self.isInOpaqueDirectory(path: $0.pathString) }
+    }
+
+    static let opaqueDirectoriesExtensions: Set<String> = {
+        [
+            "xcassets",
+            "scnassets",
+            "xcdatamodeld",
+            "docc",
+            "playground",
+            "bundle",
+        ]
+    }()
+
+    static func isInOpaqueDirectory(path: String) -> Bool {
```

Comment:
is there a reason why not to convert `path` from `String` to `AbsolutePath`, so we could leverage methods on top of it instead of parsing the string here?

Reply:
would something like this work? instead of it being a `FileManager` helper, it could be an extension on `AbsolutePath` that does the filtering?

e.g.

```swift
private let opaqueDirectories = [
    "xcassets",
    "scnassets",
    "xcdatamodeld",
    "docc",
    "playground",
    "bundle",
]
extension Array where Element == AbsolutePath {
    public func excludingOpaqueDirectoryNestedFiles() -> [AbsolutePath] {
        var encounteredOpaqueDirectories = Set<AbsolutePath>()
        var result: [AbsolutePath] = []
        forEach { path in
            let components = path.components
            let opaqueDirectoryIndex = components.firstIndex { component in
                opaqueDirectories.contains { ext in
                    component.hasSuffix(".\(ext)")
                }
            }
            if let foundIndex = opaqueDirectoryIndex {
                let opaqueDirectoryPathString = components[components.startIndex...foundIndex].joined(separator: "/")
                let opaqueDirectoryPath = AbsolutePath(opaqueDirectoryPathString)
                if !encounteredOpaqueDirectories.contains(opaqueDirectoryPath) {
                    result.append(opaqueDirectoryPath)
                }
                encounteredOpaqueDirectories.insert(opaqueDirectoryPath)
            } else {
                result.append(path)
            }
        }

        return result
    }
}

```

Need to think of how to optimise this as it will be on a "hot" code path! _(all paths will get processed via this)_

Reply:
Initially, I tried something similar to @kwridan , but found it not that performant and still complex. Just found a new approach that has similar performance (not the same, but very similar) to the code I wrote but using AbsolutePath, will push it soon.